app/war/battle.py

```diff
@@ -0,0 +1,12 @@
+from app.warrior.knight import Knight
+
+
+def start_battle(first_knight: Knight, second_knight: Knight) -> None:
+    first_knight.hp -= second_knight.power - first_knight.protection
+    second_knight.hp -= first_knight.power - second_knight.protection
```

Comment:
The calculation for reducing the knights' hp does not account for cases where the opponent's power exceeds the knight's protection, potentially resulting in negative hp values. Consider adding a check to ensure that the hp reduction does not result in a negative value, or handle this scenario appropriately.